### PR TITLE
[GEOS-10973] DWITHIN delegation to mongoDB

### PR DESCRIPTION
**Pull request by Mike7311**
[GEOS-10973](https://osgeo-org.atlassian.net/browse/GEOS-10973)
  
Added delegation of dwithin filter in cql to mongoDB for mongo schemaless plugin

Relies on [GEOT-7360](https://osgeo-org.atlassian.net/browse/GEOT-7360) DWITHIN support for mongo schemaless
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/web/core/src/main/java/org/geoserver/web/data/layer/AttributeEditPage.html [hunk 1]
```diff
@@ -19,10 +19,11 @@
     </div>
     <div wicket:id="sizeContainer" class="mb-3">
       <label for="size"><wicket:message key="size">size</wicket:message></label>
-      <input id="size" class="text" type="text" wicket:id="size"/></div>
+      <input id="size" class="text" type="text" wicket:id="size"/>
+    </div>
     <div wicket:id="crsContainer" class="mb-3">
       <label for="crs"><wicket:message key="crs">cRS</wicket:message></label>
-      <span wicket:id="crs"></span></div>
+      <span wicket:id="crs"></span>
```
Comment (Mike7311) [inline src/web/core/src/main/java/org/geoserver/web/data/layer/AttributeEditPage.html:26]: When creating new feature type from ui (when publishing new layer) it crashed due to 1 more closing tag then needed

**Comment by Mike7311** [comment]
Needs changes from https://github.com/geotools/geotools/pull/4285

**Comment by aaime** [comment]
I don't have a way to test this myself, but changes look reasonable, and the module is a community one (plus I know @Mike7311 tested manually on his end). So... merging.

**Comment by aaime** [comment]
The backport to `2.22.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply fa5939b8a3... [GEOS-10973] DWITHIN delegation to mongoDB
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/web/core/src/main/java/org/geoserver/web/data/layer/AttributeEditPage.html
CONFLICT (content): Merge conflict in src/web/core/src/main/java/org/geoserver/web/data/layer/AttributeEditPage.html

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.22.x 2.22.x
# Navigate to the new working tree
cd .worktrees/backport-2.22.x
# Create a new branch
git switch --create backport-6834-to-2.22.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick fa5939b8a3453ec6f1506ae41708a5a8f4331606
# Push it to GitHub
git push --set-upstream origin backport-6834-to-2.22.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.22.x
```
Then, create a pull request where the `base` branch is `2.22.x` and the `compare`/`head` branch is `backport-6834-to-2.22.x`.

**Pull request merged**